# HG changeset patch
# User Philip Nienhuis <prnienhuis@user.sf.net>
# Date 1519375738 -3600
#      Fri Feb 23 09:48:58 2018 +0100
# Branch stable
# Node ID e098b0fb11abfcaa4b0ae7d37175f3e71277f9f8
# Parent  9a326bc6d3152d0841c2c959810e96b9429d8ce3
inputdlg.m: Avoid crash when prompt and defaults sizes differ (bug #53209).

* inputdlg.m: Validate numel() of prompt and defaults match or emit an error.

diff --git a/scripts/gui/inputdlg.m b/scripts/gui/inputdlg.m
--- a/scripts/gui/inputdlg.m
+++ b/scripts/gui/inputdlg.m
@@ -92,16 +92,19 @@ function cstr = inputdlg (prompt, vararg
 
   linespec = 1;
   if (nargin > 2)
     linespec = varargin{2};
   endif
 
   defaults = cellstr (cell (size (prompt)));
   if (nargin > 3)
+    if (numel (varargin{3}) != numel (prompt))
+      error ("inputdlg: number of DEFAULT items must match number of PROMPT items");
+    endif
     defaults = varargin{3};
   endif
 
   if (nargin > 4)
     warning ("inputdlg: 5th 'options' argument ignored");
   endif
 
   ## specification of text field sizes as in Matlab
